.github/workflows/ci.yml

```diff
@@ -0,0 +1,111 @@
+
+on:
+  pull_request:
+    branches: ['*']
+  push:
+    branches: ['*']
+    tags: [v*]
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  build:
+    name: Build and Test
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        scala: [2.11.12, 2.12.13, 2.13.5]
+        java: [adopt@1.8]
+        platform: [jvm, js, native]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout current branch (full)
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java and Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Cache sbt
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.sbt
+            ~/.ivy2/cache
+            ~/.coursier/cache/v1
+            ~/.cache/coursier/v1
+            ~/AppData/Local/Coursier/Cache/v1
+            ~/Library/Caches/Coursier/v1
+          key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - run: sudo apt install clang libunwind-dev libgc-dev libre2-dev
+
+      - name: Check that workflows are up to date
+        run: sbt ++${{ matrix.scala }} githubWorkflowCheck
+
+      - name: Validate JVM
+        if: matrix.platform == 'jvm'
+        run: sbt ++${{ matrix.scala }} validateJVM
+
+      - name: Validate JavaScript
+        if: matrix.platform == 'js'
+        run: sbt ++${{ matrix.scala }} validateJS
+
+      - name: Validate Scala Native
+        if: matrix.platform == 'native'
+        run: sbt ++${{ matrix.scala }} validateNative
+
+  publish:
+    name: Publish Artifacts
+    needs: [build]
+    if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/v'))
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        scala: [2.13.5]
+        java: [adopt@1.8]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout current branch (full)
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java and Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Cache sbt
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.sbt
+            ~/.ivy2/cache
+            ~/.coursier/cache/v1
+            ~/.cache/coursier/v1
+            ~/AppData/Local/Coursier/Cache/v1
+            ~/Library/Caches/Coursier/v1
+          key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - uses: olafurpg/setup-gpg@v3
+
+      - env:
+          PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
+        run: sbt ++${{ matrix.scala }} ci-release
```

Comment:
Will this publish all Scala versions? It looks to me like it won't.

Reply:
Ok I see we did `sbt ci-release` before as well.
`++${{ matrix.scala }}` confused me 🙈

Reply:
To be completely honest, I'm not sure. I want to merge this and see how it behaves :shrug:

I thought https://github.com/djspiewak/sbt-github-actions/pull/56 looked like a reasonable complaint too ... let's see how it goes :smile: